zeus/tests/test_admin.py

```diff
@@ -110,3 +136,135 @@ def test_get_with_superadmin(self):
+
+    def test_find_elections(self):
+        from zeus.views.admin import find_elections
+
+        self.admin.superadmin_p = True
+        self.admin.save()
+        self.login()
+
+        election_a = self.create_election(
+            name='election A',
+            voting_starts_at_0=today_plus_days(days=1).strftime('%Y-%m-%d'),
+            voting_ends_at_0=today_plus_days(days=2).strftime('%Y-%m-%d')
+        )
+        election_b = self.create_election(
+            name='election B',
+            voting_starts_at_0=today_plus_days(days=4).strftime('%Y-%m-%d'),
+            voting_ends_at_0=today_plus_days(days=5).strftime('%Y-%m-%d')
+        )
+        election_c = self.create_election(
+            name='election C',
+            trial=True
+        )
+        election_a
+        for i, e in enumerate([election_a, election_b, election_c]):
+            e.completed_at = today_plus_days(days=6+i)
+            e.save()
+
+        # in the default case elections should be sorted by created_at, desc
+        request = RequestFactory().get(reverse('elections_report_csv'), {})
+        request.user = self.admin
+        elections = find_elections(request)
+        assert len(elections) == 2
+        assert elections[0] == election_b
+        assert elections[1] == election_a
+
+        request = RequestFactory().get(reverse('elections_report_csv'), {
+            'order': 'name',
+            'order_type': 'asc',
+        })
+        request.user = self.admin
+        elections = find_elections(request)
+        assert len(elections) == 2
+        assert elections[0] == election_a
+        assert elections[1] == election_b
+
+        request = RequestFactory().get(reverse('elections_report_csv'), {
+            'start_date': today_plus_days(days=3).strftime('%d %b %Y'),
+        })
+        request.user = self.admin
+        elections = find_elections(request)
+        assert len(elections) == 1
+        assert elections[0] == election_b
+
+        request = RequestFactory().get(reverse('elections_report_csv'), {
+            'end_date': today_plus_days(days=3).strftime('%d %b %Y'),
+        })
+        request.user = self.admin
+        elections = find_elections(request)
+        assert len(elections) == 1
+        assert elections[0] == election_a
+
+    def test_elections_report_csv(self):
+        self.admin.superadmin_p = True
+        self.admin.save()
+        self.login()
+
+        election = self.create_election(
+            name='election',
+            voting_starts_at_0=today_plus_days(days=1).strftime('%Y-%m-%d'),
+            voting_ends_at_0=today_plus_days(days=2).strftime('%Y-%m-%d')
+        )
+        election.completed_at = today_plus_days(days=3)
+        election.save()
+
+        response = self.c.get(reverse('elections_report_csv'), {})
+        date = datetime.date.today().strftime('%Y-%m-%d')
+        # make sure that the headers are right and the response body contains
```

Comment:
How about parsing the response with `csv.reader`? You could add an assertion for an exact row.